### PR TITLE
pa_ppx: minimal changes for ocaml 5.0.0 support

### PR DESCRIPTION
**Pull request by chetmurthy**
Mostly cleaning up tests

### CODE REVIEW

**Comment by chetmurthy** [comment]
Looks like everything passed except for:
1. "extras" -- due to the git repo not being available ?
2. macos/homebrew -- b/c still no support for installing perl package prereqs

**Comment by kit-ty-kate** [comment]
> "extras" -- due to the git repo not being available ?

This was simply a bug our detection of an issue in CI. If you look at the log you can see that it is because your package does not handle `ocamlopt` not being available:
```
# make[3]: Entering directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04/ocaml_stuff/5.0.0/utils'
# ocamlopt  -c pconfig.ml
# make[3]: ocamlopt: No such file or directory
# make[3]: *** [Makefile:25: pconfig.cmx] Error 127
# make[3]: Leaving directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04/ocaml_stuff/5.0.0/utils'
# make[2]: Leaving directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04/ocaml_stuff'
# make[2]: Entering directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04/lib'
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o ploc.ppo ploc.ml
# ocamlrun ../boot/camlp5r -nolib -I ../boot pa_macro.cmo -mode S -o versdep.ppo versdep.ml
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o extfun.ppo extfun.ml
# ocamlrun ../boot/camlp5r -nolib -I ../boot -mode S -o fstream.ppo fstream.ml
# ocamlfind ocamlopt -package compiler-libs,camlp-streams -g -warn-error +A-11 -c -impl extfun.ppo
# ocamlfind ocamlopt -package compiler-libs,camlp-streams -g -warn-error +A-11 -c -impl fstream.ppo
# ocamlfind: Not supported in your configuration: ocamlopt
# make[2]: *** [../config/Makefile:31: fstream.cmx] Error 2
# make[2]: *** Waiting for unfinished jobs....
# ocamlfind: Not supported in your configuration: ocamlopt
# make[2]: *** [../config/Makefile:31: extfun.cmx] Error 2
# ocamlfind ocamlopt -package compiler-libs,camlp-streams -g -warn-error +A-11 -c -impl ploc.ppo
# ocamlfind: Not supported in your configuration: ocamlopt
# make[2]: *** [../config/Makefile:31: ploc.cmx] Error 2
# ocamlfind ocamlopt -package compiler-libs,camlp-streams -g -warn-error +A-11 -c -impl versdep.ppo
# ocamlfind: Not supported in your configuration: ocamlopt
# make[2]: *** [../config/Makefile:31: versdep.cmx] Error 2
# make[2]: Leaving directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04/lib'
# make[1]: *** [Makefile:25: opt] Error 2
# make[1]: Leaving directory '/home/opam/.opam/5.0/.opam-switch/build/camlp5.8.00.04'
# make: *** [Makefile:162: world.opt] Error 2
```

File: packages/pa_ppx/pa_ppx.0.10/opam [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  "sexplib" { with-test & >= "v0.14.0" }
+  "ppx_import" { with-test & >= "1.7.1" }
+  "ppx_deriving" { with-test }
+  "ppx_deriving_yojson" { with-test & >= "3.5.2" }
+  "ppx_here" { with-test & >= "v0.13.0" }
+  "ppx_sexp_conv" { with-test & >= "v0.13.0" }
+#  "expect_test_helpers" { with-test & >= "v0.13.0" }
+]
+conflicts: [
+  "base-effects"
```
Comment (kit-ty-kate) [inline packages/pa_ppx/pa_ppx.0.10/opam:59]:
```suggestion
  "base-effects"
  "ocaml-option-bytecode-only"
```

**Comment by kit-ty-kate** [comment]
Thanks. Could you return the fix to the opam file upstream?

**Comment by chetmurthy** [comment]
Roger that, and thank you for the fixup!

**Pull request merged**